platform/darwin/test/MGLOfflineStorageTests.mm

```diff
@@ -33,32 +33,41 @@ + (void)tearDown {
     XCTAssertFalse([[NSFileManager defaultManager] fileExistsAtPath:cacheURL.path], @"Cache subdirectory should not exist.");
 }
 
+- (MGLOfflineStorage *)syncedOfflineStorage {
+    MGLOfflineStorage *os = nil;
+    @synchronized (self) {
```

Comment:
I think this should be `@synchronized ([MGLOfflineStorage class])`, given that a new instance of `MGLOfflineStorageTests` is created for each test.

Reply:
Right but each test uses a different path so there's no need to sync the class.

Reply:
Then the question is what is the `@synchronized (self)` for, given that `self` will be different for each test?